arch/src/aarch64/gic.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{boxed::Box, result};
+
+use kvm_ioctls::{DeviceFd, VmFd};
+
+use super::gicv2::GICv2;
+use super::gicv3::GICv3;
```

Comment:
Commit message updated. Thanks @sboeuf for reminding.

The reason why we implemented both GICv2 and GICv3 is that the hardware implementation in each server may be different, and some of them have only GICv2. Therefore if we don't implement GICv2, running cloud-hypervisor on such machines will have some problems.

Here in [gic.rs#L143](https://github.com/cloud-hypervisor/cloud-hypervisor/pull/1223/files#diff-f118ac49b4f181a617cc7a5d032da468R143), we try to create a GICv3 firstly, if there is any failure we then create a GICv2 instead.

Reply:
Thanks for the details, this was helpful. Also, I now realize that you rely on KVM to create these GIC devices. Maybe that's something else you can add to the commit message since the GICv2 and GICv3 emulation is really done by the host kernel through KVM.

Reply:
Done. Commit message updated.